tools/commands/integration_test.py

```diff
@@ -128,32 +126,136 @@ def run_integration_test(self, plugin_name, directory, timeout):
         else:
             # match.group(2) == 'Some tests failed.'
             return TestResult.fail(plugin_name, self.target_tuple, [
-                'flutter-tizen test integration_test failed, \
-                    see the output above for details.'
+                'flutter-tizen test integration_test failed, see the output above for details.'
             ])
 
 
+class EphemeralTarget(Target):
+    """A Tizen emulator that launches/poweroffs itself during test."""
+
+    def __init__(self, name, platform):
+        super().__init__(name, platform)
+
+    def run_integration_test(self, plugin_name, directory, timeout):
+        self.launch()
+        result = super().run_integration_test(plugin_name, directory, timeout)
+        self.power_off()
+        return result
+
+    def _find_id(self):
+        completed_process = subprocess.run('sdb devices',
+                                           shell=True,
+                                           cwd='.',
+                                           universal_newlines=True,
+                                           stderr=subprocess.PIPE,
+                                           stdout=subprocess.PIPE)
+        if completed_process.returncode != 0:
+            raise Exception('sdb failure.')
+
+        lines = completed_process.stdout.rstrip().split('\n')
+        for line in lines[1:]:
+            tokens = re.split('[\t ]', line)
+            id = tokens[0]
+            name = tokens[-1]
+            if name == self.name:
+                return id
+        raise Exception(f'Could not find connected target {self.name}')
+
+    def launch(self):
+        completed_process = subprocess.run(f'em-cli launch -n {self.name}',
+                                           shell=True,
+                                           cwd='.')
+        if completed_process.returncode != 0:
+            raise Exception(f'Target {self.name} launch failed.')
+        time.sleep(5)
+        self.id = self._find_id()
+        pass
+
+    def power_off(self):
+        # The poweroff logic for tv is different because sdb root is not allowed.
+        if self.device_profile == 'tv':
+            completed_process = subprocess.run(
```

Comment:
why don't you save and use the pid of the launched emulator instead of finding the pid by using ps and grep?

Reply:
Just so that we're on the same page, do you mean creating an instance variable `self.pid` in `EphemeralTarget.launch()` and using is later in `EphemeralTarget.power_off()`?

Reply:
yes I think it's more clear if possible.

Reply:
OK, I'll change it.